pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -126,11 +126,21 @@ func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(pt.validateRefOrSpec())
 
 	errs = errs.Also(pt.validateEmbeddedOrType())
-
+	// taskKinds contains the kinds when the apiVersion is not set, they are not custom tasks,
+	// if apiVersion is set they are custom tasks.
+	taskKinds := map[TaskKind]bool{
+		"":                 true,
+		NamespacedTaskKind: true,
+		"ClusterTask":      true,
```

Comment:
we don't have cluster tasks in v1
```suggestion
```

Reply:
I think this might not have been correct: https://github.com/tektoncd/pipeline/issues/6587